docs/autoprocess.py

```diff
@@ -0,0 +1,412 @@
+        # the order in which they are displayed
+        property_names = [
+            'label',
+            'type',
+            'description',
+            'required',
+            'disabled',
+            'hidden',
+            'default',
+            'placeholder',
+            'validate_regex',
+            'choices',
+            'collapse',
+            'group']
+
+        for name in property_names:
+            if name in data:
+                value = data.pop(name)
+
+                # Value should be formatted in code-style (=literal) mode
+                if name in ['type', 'default', 'placeholder', 'validate_regex']:
+                    literal_node = nodes.literal(str(value), str(value))
+                    properties_list += self.make_field(name, literal_node)
+
+                # Special formating of `value` is needed if name == 'choices'
+                elif name == 'choices':
+                    bullet_list = nodes.bullet_list()
+                    for choice in value:
+                        list_item = nodes.list_item()
+                        paragraph = nodes.paragraph()
+                        label = nodes.Text(choice['label'] + ': ')
+                        val = nodes.literal(choice['value'], choice['value'])
+
+                        paragraph += label
+                        paragraph += val
+                        list_item += paragraph
+                        bullet_list += list_item
+
+                    properties_list += self.make_field(name, bullet_list)
+
+                # Make a nested list of properties:
+                elif name == 'group':
+                    list123 = nodes.field_list()
+                    for sub_param in value:
+                        sub_param_properties_list = self.make_properties_list(sub_param)
+                        list123 += self.make_field(sub_param['name'], sub_param_properties_list)
+
+                    properties_list += self.make_field(name, list123)
+
+                else:
+                    properties_list += self.make_field(name, str(value))
+
+        return properties_list
+
+    def make_process_node(self, github_url, process):
+        """
+        Fill the content of process definiton node
+
+        :param github_base_url: url to process definition on GitHub
+        :type github_base_url: url/string
+        :param process: process data as given from yaml.load function
+        :type process: dict
+
+        :rtype: nodes.field
+        """
+
+        name = process['name']
+
+        # Make process name as section title:
+        section = nodes.section()
+        section['ids'] = [str(uuid.uuid4())]
+        title = nodes.subtitle()
+        title_text = nodes.Text(name)
+        title += title_text
+        section += title
+
+        top_field_list = nodes.field_list()
+
+        slug = process['slug']
+        version = process['version']
+        description = process['description']
```

Comment:
There is a chance that `description` is not given.

    description = process.get('description', '')